agent/dns.go

```diff
@@ -938,6 +955,19 @@ func (d *DNSServer) preparedQueryLookup(network, datacenter, query string, req,
 			Node:       d.agent.config.NodeName,
 		},
 	}
+	
+	subnet := ednsSubnetForRequest(req)
+	
+	if subnet != nil {
+		args.Source.Ip = subnet.Address.String()
+	} else {
+		switch v := remoteAddr.(type) {
+			case *net.TCPAddr:
+				args.Source.Ip = v.IP.String()				
+			case *net.UDPAddr:
+				args.Source.Ip = v.IP.String()
```

Comment:
Is it possible for remoteAddr to come in as a different type? I think net.IPAddr and maybe some others are net.Addr but I don't know if that's relevant here.

Reply:
I didn't think it was relevant. The remoteAddr here comes from the dns lib which sets it specifically to a UDPAddr or by calling the RemoteAddr method on a net.Conn instance it gets from its TCPListener. The assumption here was that a TCPListener should have a RemoteAddr returning a TCPAddr. Maybe not. Either way its easy enough to add a IPAddr case.